jupyterhub/templates/proxy/service.yaml

```diff
@@ -44,26 +43,18 @@ spec:
     {{- if $HTTPS }}
     - name: https
       port: 443
-      protocol: TCP
-      {{- if or $manualHTTPS $manualHTTPSwithsecret }}
-      targetPort: 8443
-      {{- else if $offloadHTTPS }}
-      targetPort: 8000
+      {{- if $offloadHTTPS }}
+      targetPort: http
       {{- else }}
-      targetPort: 443
+      targetPort: https
```

Comment:
No matter if this goes to the deploy/autohttps or deploy/proxy, it should always go to the `https` port unless the "offload".